presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -901,6 +904,11 @@ public void endAnalysis()
         queryStateTimer.endAnalysis();
     }
 
+    public void setResourceGroupQueuedOn(Optional<ResourceGroupId> resourceGroup)
```

Comment:
This is not thread-safe

Reply:
Updated to make `resourceGroupQueuedOn` atomic.